### PR TITLE
Pull request for metricd-not-implemented-sacks

### PR DESCRIPTION
**Pull request by jd**
metricd: fix coordinator not implemented error
metricd: be specific about tooz.NotImplemented

### CODE REVIEW

**Comment by chungg** [comment]
this is fine, although i'd like to see us fix the notifier feature first... or revert and defer to 4.2 if we can't fix it.

**Comment by jd** [comment]
@chungg lol remain me last time we did not fix something

**Comment by chungg** [comment]
:) i'm just saying. you mentioned you wanted to get 4.1 release out ASAP last week and i'm not sure how simple the fix is tbh.

**Comment by jd** [comment]
Haha sure. Anyway that should not prevent this to be merged, it's a correct fix AFAICT.

**Comment by chungg** [comment]
yeah, it is. but if we do defer, i'm saving us having to revert multiple patches.

**Comment by jd** [comment]
That's a sad and pessimistic reason.

File: gnocchi/cli/metricd.py [hunk 1]
```diff
@@ -187,6 +187,11 @@ def _get_sacks_to_process(self):
                     i for i in six.moves.range(self.incoming.NUM_SACKS)
                     if self.partitioner.belongs_to_self(
                         i, replicas=self.conf.metricd.processing_replicas)]
+        except tooz.NotImplemented:
```
Comment (chungg) [inline gnocchi/cli/metricd.py]: you going to move the run_watchers command elsewhere? do we still need this if you do?

**Comment by jd** [comment]
I'd prefer we merge it for now as it avoids having metricd yelling. We can fix run_watchers after. Feel free to open a bug and target to milestone 4.1 if you're scared we forget :)

**Pull request merged**